service/topology_coordinator.cc

```diff
@@ -84,6 +86,13 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
 
     tablet_allocator& _tablet_allocator;
 
+    // The reason load_stats_ptr is a shared ptr is that load balancer can yield, and we don't want it
+    // to suffer lifetime issues when stats refresh fiber overrides the current stats.
+    locator::load_stats_ptr _tablet_load_stats;
+    // FIXME: make frequency per table in order to reduce work in each iteration.
+    //  Bigger tables will take longer to be resized. similar-sized tables can be batched into same iteration.
```

Comment:
If we're doing this per-node and not per-table-per-node, then this is outdated, no?

Reply:
I thought you still wanted to reduce work in the load balancer by reducing the frequency we check need for resize decision based on the table size, not only the amount of work in the replicas, but also in the coordinator. Might be relevant with a large table count.

Reply:
Yes. I guess we could change the frequency based on the smallest table. Or something. A fixed frequency will carry us a long way, this is a very small optimization.